pkgs/servers/misskey/default.nix

```diff
@@ -0,0 +1,95 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchYarnDeps
+, fixup_yarn_lock
+, yarn
+, nodejs
+, python3
+, pkg-config
+, glib
+, vips
+}:
+
+let
+  version = "12.107.0";
+
+  src = fetchFromGitHub {
+    owner = "misskey-dev";
+    repo = "misskey";
+    rev = version;
+    sha256 = "sha256-ovoqqFnWZe99yeYWKfiGR0BgyvOnWyb2wa026qKL8Io=";
+  };
+
+  deps = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    sha256 = "sha256-c8R5w0DffBtEkaePyPjJsb3OsNaN9lCDzf4QWSTfMPk=";
+  };
+  backendDeps = fetchYarnDeps {
+    yarnLock = "${src}/packages/backend/yarn.lock";
+    sha256 = "sha256-AJ9RB0fJTHELPoqshM9Tvaq/fXpwiiU55n4zvSvZRN4=";
+  };
+  clientDeps = fetchYarnDeps {
+    yarnLock = "${src}/packages/client/yarn.lock";
+    sha256 = "sha256-Jk3XCsrSUyRAaUC6tJBqBslmU2J26bJyw64TAovCLH8=";
+  };
+
+in stdenv.mkDerivation {
+  pname = "misskey";
+  inherit version src;
+
+  nativeBuildInputs = [ fixup_yarn_lock yarn nodejs python3 pkg-config ];
+  buildInputs = [ glib vips ];
+
+  buildPhase = ''
+    export HOME=$PWD
+    export NODE_ENV=production
+
+    # Build node modules
+    fixup_yarn_lock yarn.lock
+    fixup_yarn_lock packages/backend/yarn.lock
+    fixup_yarn_lock packages/client/yarn.lock
+    yarn config --offline set yarn-offline-mirror ${deps}
+    yarn install --offline --frozen-lockfile --ignore-engines --ignore-scripts --no-progress
+    (
+      cd packages/backend
+      yarn config --offline set yarn-offline-mirror ${backendDeps}
+      yarn install --offline --frozen-lockfile --ignore-engines --ignore-scripts --no-progress
+    )
+    (
+      cd packages/client
+      yarn config --offline set yarn-offline-mirror ${clientDeps}
+      yarn install --offline --frozen-lockfile --ignore-engines --ignore-scripts --no-progress
+    )
+    patchShebangs node_modules
+    patchShebangs packages/backend/node_modules
+    patchShebangs packages/client/node_modules
+    (
+      cd packages/backend/node_modules/re2
+      npm_config_nodedir=${nodejs} npm run rebuild
+    )
+    (
+      cd packages/backend/node_modules/sharp
+      npm_config_nodedir=${nodejs} ../.bin/node-gyp rebuild
+    )
+
+    yarn build
+  '';
+
+  installPhase = ''
+    mkdir -p $out/packages/client
+    ln -s /var/lib/misskey $out/files
+    ln -s /etc/misskey $out/.config
```

Comment:
Yes, I thought about that, but wasn't there some security concerns with envsubst because normal users might be able to read the process' environment from /proc?

Reply:
Nevermind that was about passing the arguments to sed or the likes on the command line.
I will change this to use envsubst.

Reply:
This works great, thanks.